snare/cloner.py

```diff
@@ -31,11 +31,16 @@ def __init__(self, root, max_depth, css_validate):
 
     @staticmethod
     def add_scheme(url):
+        test_string = str(url)
         if yarl.URL(url).scheme:
             new_url = yarl.URL(url)
-            err_url = yarl.URL(url + '/status_404')
+            err_url = yarl.URL(url + 'status_404')
         else:
             new_url = yarl.URL('http://' + url)
+            err_url = yarl.URL('http://' + url + 'status_404')
+        if test_string[-1] == '/':
```

Comment:
It will produce wrong results. Move this condition at start and just remove the trailing `/`